x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoGridTilerTests.java

```diff
@@ -106,7 +106,7 @@ public void testGeoTileSetValuesBruteAndRecursivePoints() throws Exception {
     // tests that bounding boxes of shapes crossing the dateline are correctly wrapped
     public void testGeoTileSetValuesBoundingBoxes_BoundedGeoShapeCellValues() throws Exception {
         for (int i = 0; i < 1; i++) {
-            int precision = randomIntBetween(0, 4);
+            int precision = randomIntBetween(1, 4);
```

Comment:
I notice that this test might fail for precision 0. That precision is handle by AllCellValues so let's remove it.

Reply:
Do we have a separate test for that? Maybe it's better to pull GeoShapeCellIdSource in here and make sure that everything is still covered?

Reply:
I am going to push it as it is. I handle this case in a separate PR as I think the current implementation is not right.